components/InfoPane/FilePicker.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { readFile } from '../../lib/read_file.js';
+
+// Handle file selection via input
+function onSelect(ev: React.ChangeEvent<HTMLInputElement>) {
+  const file = ev.target.files?.item(0);
+  if (file) {
+    readFile(file);
+  }
+
+  // Reset field
+  ev.target.value = '';
+}
+
+export default function FilePicker({ label }: { label: string }) {
+  return (
+    <label className="link" style={{ display: 'inline' }}>
+      {label}
+      <input type="file" hidden onChange={onSelect} accept=".json" />
+    </label>
```

Comment:
New file-picker-only component. It appears as the single "Choose file" _link_

<img width="306" alt="Screenshot" src="https://github.com/user-attachments/assets/07b4e7fc-0afe-4ff3-b1b6-e0f2c3d15a61">